LAPACKE dgesvd appears check ldvt too strictly in the row major case

While looking into a Gonum user issue (gonum/gonum#2008), I found that when calling dgesvd with memory allocated by our wrapper for the JOBU/JOBV N case results in a LAPACKE_xerbla, "Wrong parameter 12 in LAPACKE_dgesvd_work". This appears to be happening due to `LAPACKE_dgesvd_work` unconditionally checking leading dimensions of the output `VT` matrix even in the case that it will not be used.

This happens [here](https://github.com/OpenMathLib/OpenBLAS/blob/74be676936b22eacccd88fc7e19cfb224625c77b/lapack-netlib/LAPACKE/src/lapacke_dgesvd_work.c#L75-L79) and [here](https://github.com/OpenMathLib/OpenBLAS/blob/74be676936b22eacccd88fc7e19cfb224625c77b/lapack-netlib/LAPACKE/src/lapacke_dgesvd_work.c#L56-L57)

The [netlib LAPACK reference for DGESVD](https://netlib.org/lapack/explore-html/d1/d7f/group__gesvd_gac6bd5d4e645049e49bb70691180abf07.html) states
> LDVT is INTEGER
> The leading dimension of the array VT.  LDVT >= 1; if
> JOBVT = 'A', LDVT >= N; if JOBVT = 'S', LDVT >= min(M,N).

Indicating that it is unconstrained in the `N` case.

A Gonum-dependent replicator is available at gonum/gonum#2008, but also simplified here for convenience.
```
package main

import (
	"math/rand"

	"gonum.org/v1/gonum/blas/blas64"
	"gonum.org/v1/gonum/lapack/lapack64"
	"gonum.org/v1/gonum/mat"
	netlib_blas "gonum.org/v1/netlib/blas/netlib"
	netlib_lapack "gonum.org/v1/netlib/lapack/netlib"
)

func main() {
	blas64.Use(netlib_blas.Implementation{})
	lapack64.Use(netlib_lapack.Implementation{})

	data := make([]float64, 10*10)
	for i := range data {
		data[i] = rand.Float64()
	}

	a := mat.NewDense(10, 10, data)
	a.Mul(a.T(), a)

	var svd mat.SVD
	svd.Factorize(a, mat.SVDNone) // Does not fail with mat.SVDFull.
}
```
outputs
```
Wrong parameter 12 in LAPACKE_dgesvd_work
panic: lapack: insufficient declared workspace length

goroutine 1 [running]:
gonum.org/v1/netlib/lapack/netlib.Implementation.Dgesvd({}, 0x48?, 0xf2?, 0x10?, 0x49ac28?, {0xc0000ae380?, 0x10?, 0xa?}, 0xa?, {0xc000022140, ...}, ...)
	…/pkg/mod/gonum.org/v1/netlib@v0.0.0-20230729102104-8b8060e7531f/lapack/netlib/lapack.go:2019 +0x44f
gonum.org/v1/gonum/lapack/lapack64.Gesvd(0x4e, 0x4e, {0xa, 0xa, {0xc0000ae380, 0x64, 0x64}, 0xa}, {0x0, 0x0, ...}, ...)
	…/pkg/mod/gonum.org/v1/gonum@v0.15.1/lapack/lapack64/lapack64.go:354 +0x1d0
gonum.org/v1/gonum/mat.(*SVD).Factorize(0xc000073e98, {0x5580a8, 0xc00003a100}, 0x0)
	…/pkg/mod/gonum.org/v1/gonum@v0.15.1/mat/svd.go:139 +0x8f4
main.main()
	…/thinking/netlib/main.go:26 +0x1ee
exit status 2
```
The Go panic occurs because the first call to `dgesvd` to get the work size returns in error and so does not indicate the correct work length for the second call. The client code that is used here [is unsurprising use of dgesvd](https://github.com/gonum/gonum/blob/f42c07e8cbea534b003124f60cf48ebebe5fec2e/mat/svd.go#L79-L145).

For reference, the Gonum dimension checks are [here](https://github.com/gonum/netlib/blob/8b8060e7531fd05ff4f0e7f096fee4884e595718/lapack/netlib/lapack.go#L1989-L2022).

Looking at the other checks in lapacke_dgesvd_work.c, I think that the [lda check](https://github.com/OpenMathLib/OpenBLAS/blob/74be676936b22eacccd88fc7e19cfb224625c77b/lapack-netlib/LAPACKE/src/lapacke_dgesvd_work.c#L65) could be improved to `if( lda < MAX(1, n) ) {`, though this is not causing a problem here.

--- a/lapack-netlib/LAPACKE/src/lapacke_cgesvd_work.c
+++ b/lapack-netlib/LAPACKE/src/lapacke_cgesvd_work.c
@@ -74,10 +74,12 @@ lapack_int LAPACKE_cgesvd_work( int matrix_layout, char jobu, char jobvt,
             LAPACKE_xerbla( "LAPACKE_cgesvd_work", info );
             return info;
         }
-        if( ldvt < ncols_vt ) {
+	if( LAPACKE_lsame( jobvt, 'a' ) || LAPACKE_lsame( jobvt, 's' ) ) {
+	if( ldvt < ncols_vt ) {
             info = -12;
             LAPACKE_xerbla( "LAPACKE_cgesvd_work", info );
             return info;
+        }
         }
         /* Query optimal working array(s) size if requested */
         if( lwork == -1 ) {

--- a/lapack-netlib/LAPACKE/src/lapacke_dgesvd_work.c
+++ b/lapack-netlib/LAPACKE/src/lapacke_dgesvd_work.c
@@ -72,10 +72,12 @@ lapack_int LAPACKE_dgesvd_work( int matrix_layout, char jobu, char jobvt,
             LAPACKE_xerbla( "LAPACKE_dgesvd_work", info );
             return info;
         }
+	if( LAPACKE_lsame( jobvt, 'a' ) || LAPACKE_lsame( jobvt, 's' ) ) {
         if( ldvt < ncols_vt ) {
             info = -12;
             LAPACKE_xerbla( "LAPACKE_dgesvd_work", info );
             return info;
+        }
         }
         /* Query optimal working array(s) size if requested */
         if( lwork == -1 ) {

--- a/lapack-netlib/LAPACKE/src/lapacke_sgesvd_work.c
+++ b/lapack-netlib/LAPACKE/src/lapacke_sgesvd_work.c
@@ -72,10 +72,12 @@ lapack_int LAPACKE_sgesvd_work( int matrix_layout, char jobu, char jobvt,
             LAPACKE_xerbla( "LAPACKE_sgesvd_work", info );
             return info;
         }
-        if( ldvt < ncols_vt ) {
+        if( LAPACKE_lsame( jobvt, 'a' ) || LAPACKE_lsame( jobvt, 's' ) ) {
+	if( ldvt < ncols_vt ) {
             info = -12;
             LAPACKE_xerbla( "LAPACKE_sgesvd_work", info );
             return info;
+        }
         }
         /* Query optimal working array(s) size if requested */
         if( lwork == -1 ) {

--- a/lapack-netlib/LAPACKE/src/lapacke_zgesvd_work.c
+++ b/lapack-netlib/LAPACKE/src/lapacke_zgesvd_work.c
@@ -74,10 +74,12 @@ lapack_int LAPACKE_zgesvd_work( int matrix_layout, char jobu, char jobvt,
             LAPACKE_xerbla( "LAPACKE_zgesvd_work", info );
             return info;
         }
-        if( ldvt < ncols_vt ) {
+	if( LAPACKE_lsame( jobvt, 'a' ) || LAPACKE_lsame( jobvt, 's' ) ) {
+	if( ldvt < ncols_vt ) {
             info = -12;
             LAPACKE_xerbla( "LAPACKE_zgesvd_work", info );
             return info;
+        }
         }
         /* Query optimal working array(s) size if requested */
         if( lwork == -1 ) {

